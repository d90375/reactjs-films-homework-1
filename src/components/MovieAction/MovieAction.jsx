import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MovieDescription from '../MovieDescription';
import Button from '../Button';

import styles from './MovieAction.scss';

class MovieAction extends Component {
  state = {
    descriptionClass: 'hide',
    buttonValue: 'View Info',
  }

  clickViewHandler = () => {
    this.setState(({ descriptionClass }) => (descriptionClass === 'show'
      ? { descriptionClass: 'hide', buttonValue: 'View Info' }
      : { descriptionClass: 'show', buttonValue: 'Hide Info' }));
  }

  render() {
    const { description } = this.props;
    const { descriptionClass, buttonValue } = this.state;
    return (
      <div className={styles.container}>
        <MovieDescription descriptionClass={descriptionClass} description={description} />
        <div className={styles.buttonsContainer}>
          <Button purpose="watch">Watch Now</Button>
          <Button purpose="view" clickHandler={this.clickViewHandler}>{buttonValue}</Button>
        </div>
      </div>
    );
  }
}

MovieAction.propTypes = {
  description: PropTypes.string.isRequired,
};

export default MovieAction;
