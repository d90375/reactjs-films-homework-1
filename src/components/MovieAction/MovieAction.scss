.container {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  @media (max-width: 1024px) {
    flex-direction: column;
    align-items: center;
  }
}

.buttonsContainer {
  display: flex;
  justify-content: space-around;
  width: 250px;
  padding-top: 20px;
}
