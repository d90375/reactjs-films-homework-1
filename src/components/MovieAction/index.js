import MovieAction from './MovieAction';

export default MovieAction;
