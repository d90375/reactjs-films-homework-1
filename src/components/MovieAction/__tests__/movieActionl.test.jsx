import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create, act } from 'react-test-renderer';

import MovieAction from '../MovieAction';

test('MovieAction renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<MovieAction description="description" />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});

test('clickViewHandler function change state', () => {
  const tree = create(
    <MovieAction description="description" />,
  );

  const button = tree.root.findByProps({ 'aria-label': 'view' });
  expect(button.props.children).toBe('View Info');

  const mockFunction = jest.fn(() => button.props.clickHandler());

  act(() => {
    mockFunction();
  });

  expect(button.props.children).toBe('Hide Info');

  act(() => {
    mockFunction();
  });

  expect(button.props.children).toBe('View Info');
});
