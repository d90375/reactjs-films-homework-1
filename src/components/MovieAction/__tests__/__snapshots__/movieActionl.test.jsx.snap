// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieAction renders correctly 1`] = `
<div
  className="container"
>
  <MovieDescription
    description="description"
    descriptionClass="hide"
  />
  <div
    className="buttonsContainer"
  >
    <Button
      purpose="watch"
    >
      Watch Now
    </Button>
    <Button
      aria-label="view"
      clickHandler={[Function]}
      purpose="view"
    >
      View Info
    </Button>
  </div>
</div>
`;
